src/orchard/java.clj

```diff
@@ -168,7 +168,7 @@
                        (source-info class)
                        {:name       (-> c .getSimpleName symbol)
                         :class      (-> c .getName symbol)
-                        :package    (-> c package symbol)
+                        :package    (some-> c package symbol)
```

Comment:
I think this should be handled in the `package` function. And we also need to add a test for this. I was hoping the approach suggested by @ghadishayban [here](https://dev.clojure.org/jira/browse/CLJ-1550) was going to sort this case as well. 

We also need an unit test covering this case.

Reply:
So, `package` should return symbol or nil?

Reply:
`nil`. The current approach is fine, we just need a test for this. I never use `reify`, so I didn't think of this.